install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -625,6 +625,36 @@ spec:
         resources:
           {{- toYaml . | trim | nindent 10 }}
         {{- end }}
+      {{- if .Values.waitForKubeProxy }}
+      - name: wait-for-kube-proxy
+        image: {{ include "cilium.image" .Values.image | quote }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command:
+          - bash
+          - -c
+            - |
+              while true
+              do
+                if iptables-nft-save -t mangle | grep -E '^:(KUBE-IPTABLES-HINT|KUBE-PROXY-CANARY)'; then
```

Comment:
Instead of doing a save and grep could you use the `--check` flag to check for the chain and a particular rule?

```
       -C, --check chain rule-specification
              Check whether a rule matching the specification does exist in the selected chain. This command uses the same logic as -D to  find  a  matching
              entry, but does not alter the existing iptables configuration and uses its exit code to indicate success or failure.
```

Reply:
not sure... I've never used --check maybe @joestringer knows the answer.